source/core/parameters.cc

```diff
@@ -2504,21 +2515,36 @@ namespace Parameters
           throw(std::runtime_error(
             "Unknown verbosity mode for the eigenvalue estimation"));
 
-        mg_coarse_grid_max_iterations =
-          prm.get_integer("mg coarse grid max iterations");
-        mg_coarse_grid_tolerance = prm.get_double("mg coarse grid tolerance");
-        mg_coarse_grid_reduce    = prm.get_double("mg coarse grid reduce");
-        mg_coarse_grid_max_krylov_vectors =
-          prm.get_integer("mg coarse grid max krylov vectors");
+        const std::string cg_solver = prm.get("mg coarse grid solver");
+        if (cg_solver == "gmres")
+          mg_coarse_grid_solver = CoarseGridSolverType::gmres;
+        else if (cg_solver == "amg")
+          mg_coarse_grid_solver = CoarseGridSolverType::amg;
+        else if (cg_solver == "ilu")
+          mg_coarse_grid_solver = CoarseGridSolverType::ilu;
+        else if (cg_solver == "direct")
+          mg_coarse_grid_solver = CoarseGridSolverType::direct;
+        else
+          throw std::logic_error(
+            "Error, invalid coarse grid solver type. Choices are gmres, amg, ilu or direct.");
```

Comment:
I think these asserts are not needed, since they are already handled during parsing.

Reply:
Yeah when you think about it, @peterrum is right. I don't know why we have this habit of asserting this.

Reply:
That's a good point. So we are kind of asserting twice:  when we parse all parameters and then in the actual solver... should I then remove the asserts in the `mf_navier_stokes.cc`:

https://github.com/chaos-polymtl/lethe/blob/9a43846f67bbc159c7a63fd604884c73625f4ae2/source/solvers/mf_navier_stokes.cc#L1267